### PR TITLE
feat: add resource window

### LINKED ISSUE #82: Improve performance by providing a resource context
Flutter prints this error. The feature is actually an optional feature, but highly recommended.

`[ERROR:flutter/shell/platform/embedder/embedder_surface_gl.cc(89)] Could not create a resource context for async texture uploads. Expect degraded performance. Set a valid make_resource_current callback on FlutterOpenGLRendererConfig.`

### PR DESCRIPTION
**Pull request by pchampio**
fixes #82

### CODE REVIEW

File: application.go [hunk 1]
```diff
@@ -92,6 +106,11 @@ func (a *Application) Run() error {
 	glfw.DefaultWindowHints()
 	defer a.window.Destroy()
 
+	a.resourceWindow, err = createResourceWindow(a.window)
```
Comment (GeertJohan) [inline application.go]: This error is never checked?

File: application.go [hunk 2]
```diff
@@ -92,6 +106,11 @@ func (a *Application) Run() error {
...
+	if a.resourceWindow != nil {
```
Comment (GeertJohan) [inline application.go]: What if it is nil?

Comment (GeertJohan) [reply application.go]: Ah, I see it is handled below.. So we're being graceful here, if no window can be created, the app will work like it does currently..

Comment (GeertJohan) [reply application.go]: I think we should check the error above though. Not to exit, but just to print a message with the reason why no resource window could be created.

**Review by GeertJohan — CHANGES_REQUESTED** [review]

File: application.go (hunk 2 above)
Comment (Drakirus) [reply application.go]: Done!

**Review by GeertJohan — APPROVED** [review]

**Pull request merged**